app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx

```diff
@@ -30,31 +60,133 @@ const useRuleTableColumns = () => {
       render: (rule: RuleTableDataType) => {
         const checked = rule?.status === RuleObjStatus.ACTIVE ? true : false;
-
         return <Switch checked={checked} onChange={(checked: boolean) => handleStatusToggle([rule], checked)} />;
       },
     },
     {
-      title: "Actions",
+      title: "Updated on",
+      key: "modificationDate",
+      align: "center",
+      width: 152,
+      responsive: ["lg"],
+      render: (rule: RuleTableDataType) => {
+        if (rule.objectType && rule.objectType === RuleObjType.GROUP) {
+          return null;
+        }
+        const dateToDisplay = rule.modificationDate ? rule.modificationDate : rule.creationDate;
+        const beautifiedDate = moment(dateToDisplay).format("MMM DD");
+        if (currentlyActiveWorkspace?.id && !options?.hideLastModifiedBy) {
+          return (
+            <span>
+              {beautifiedDate} by <UserIcon uid={rule.lastModifiedBy} />
+            </span>
+          );
+        } else return beautifiedDate;
+      },
+    },
+    {
+      title: "",
       key: "actions",
-      render: (rule: RuleTableDataType) => (
-        <>
-          <Button
-            onClick={() => {
-              handleRuleShare(rule);
-            }}
-            type={"text"}
-            icon={<BsShare />}
-          />
-        </>
-      ),
+      width: 104,
+      align: "right",
+      render: (record: RuleTableDataType) => {
+        const isRule = record.objectType === RuleObjType.RULE;
+
+        const recordActions: MenuProps["items"] = [
+          {
+            key: 0,
+            onClick: () => {
+              handleRenameGroupClick(record);
+            },
+            label: (
+              <Row>
+                <RiEdit2Line /> Rename
+              </Row>
+            ),
+          },
+          {
+            key: 1,
+            onClick: () => {
+              handleDuplicateRuleClick(record);
+            },
+            label: (
+              <Row>
+                <RiFileCopy2Line />
+                Duplicate
+              </Row>
+            ),
+          },
+          {
+            key: 2,
+            danger: true,
+            onClick: () => handleDeleteRecordClick(record),
+            label: (
+              <Row>
+                <RiDeleteBinLine />
+                Delete
+              </Row>
+            ),
+          },
+        ].filter((option) => (!isRule ? option.key !== 1 : option.key !== 0));
+
+        return (
+          <Row align="middle" wrap={false} className="rules-actions-container">
+            <Button
+              type="text"
+              icon={<MdOutlineShare />}
+              onClick={() => {
+                handleRuleShare(record);
+              }}
+            />
+
+            <Dropdown menu={{ items: recordActions }} trigger={["click"]} overlayClassName="rule-more-actions-dropdown">
+              <Button type="text" className="more-rule-actions-btn" icon={<MdOutlineMoreHoriz />} />
+            </Dropdown>
```

Comment:
We should create a separate component like `RuleTableActions` which has this rendering logic and then assign this component to `render` property of this col. We may have more actions in future